# Using EDHEC Risk Kit Part 5
### Explore use of the library to:
- Use the CAPM to determine the an appropriate return of an asset
- Explore the coefficients in CAPM
- Conduct factor analysis

### CAPM
- The expected return on any asset $i$ is the risk-free interest rate, $R_F$, plus a risk premium, which is the asset's market beta, $\beta_{iM}$, times the premium per unit of beta risk, $E(R_M) - R_f$
- $E(R_i) = R_f + \beta_{iM}(E(R_m) - R_f)$
    - $\beta_{iM}$ for asset $i$ is the covariance of its return with the market return divided by the variance of the market return
    - (Market Beta) $\beta_{iM} = \frac{\text{cov}(R_i,R_M)}{\sigma^2(R_M)}$
    - Market beta is a risk premium
    - Assets with a market beta of zero must have an expected return equal to the risk free rate because they doesn't contribute to the variance in the market return

### CAPM assumptions
- Expected returns for each asset is linearly related to their market betas
- Expected return on market portfolio > expected return on assets with market beta of 0
    - Market betas for all assets are positive
- Assets uncorreleated with the market have expected returns equal $R_f$
- The beta premium is equal to $E(R_m) - R_f$

### Testing CAPM on market data
- Regress average asset returns on estimates of asset betas
- Model predicts the intercept is the riskfree rate
- Coefficient on beta is $E(R_M)-R_f$

### Issues that arise
- Tough to estimate market beta for an individual asset, can vary depending on time range
    - Estimate beta for portfolios of assets instead
- Residuals from the model for one asset are not independent from another's residuals
    - Caused by variation due to an attribute such as industry or location

In [23]:
import pandas as pd
import edhec_risk_kit as erk
import numpy as np
import statsmodels.api as sm

In [38]:
ind_rets = erk.get_ind_file('returns', n_inds=49)
fff = erk.get_fff_returns()

### What is the market beta for Healthcare industry stocks from 1970 to 2015?

In [46]:
hlth_rets = ind_rets.loc['1970':'2015', ['Hlth']]
mkt_excess = fff.loc['1970':'2015', ['Mkt-RF']]
exp_var = mkt_excess.copy()
exp_var['Constant'] = 1
lm = sm.OLS(hlth_rets, exp_var).fit()
lm.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                   Hlth   R-squared:                       0.387
Model:                            OLS   Adj. R-squared:                  0.386
Method:                 Least Squares   F-statistic:                     347.3
Date:                Fri, 13 Mar 2020   Prob (F-statistic):           1.92e-60
Time:                        14:03:13   Log-Likelihood:                 730.63
No. Observations:                 552   AIC:                            -1457.
Df Residuals:                     550   BIC:                            -1449.
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Mkt-RF         1.1177      0.060     18.635      0.000       1.000       1.236
Constant       0.0040      0.003      1.464      0.144      -0.001       0.009
==============================================================================
Omnibus:                       48.762   Durbin-Watson:                   1.818
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              250.973
Skew:                          -0.081   Prob(JB):                     3.18e-55
Kurtosis:                       6.299   Cond. No.                         21.8
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

### The market beta for healthcare stocks is the coefficient on the Mkt-RF variable in the model. In this case, it is 1.1177. This tells us that the expected excess return on healthcare stocks is 1.1177 times the excess return of the market

### Which industry had the highest market beta from 1960 to 2015?
- Consider all 49 industries

In [32]:
industries = list(ind_rets.columns)
betas = []
for ind in industries:
    ind_excess = ind_rets.loc['1960':'2015 ', ind]
    lm = sm.OLS(ind_excess, exp_var).fit()
    betas.append(lm.params['Mkt-RF'])

beta_df = pd.DataFrame({'Beta':betas},
                        index = industries)

In [33]:
print(beta_df['Beta'].idxmax(), 'had a market beta of', np.round(beta_df['Beta'].max(),2))

Chips had a market beta of 1.39


### Lowest market beta?

In [34]:
print(beta_df['Beta'].idxmin(), 'had a market beta of', np.round(beta_df['Beta'].min(),2))

Util had a market beta of 0.54
